crates/hir/src/lib.rs

```diff
@@ -3258,10 +3454,13 @@ impl Impl {
                     .filter(filter),
             )
         });
+
         for id in def_crates
             .iter()
             .flat_map(|&id| Crate { id }.transitive_reverse_dependencies(db))
             .map(|Crate { id }| id)
+            .chain(Crate { id: env.krate }.dependencies(db).into_iter().map(|it| it.krate.id))
+            .chain(std::iter::once(env.krate))
```

Comment:
Can you point me to a test that fails without these lines? I'm still a bit confused as to why this is necessary

Reply:
`handlers::typed_hole::tests::local_item_use_trait` fails without line 3463 as it doesn't see impl right next to function somehow. Starting to wonder if 3462 is necessary as I cant repr issue with types from std

Reply:
My bad, I'll change the test to not use primitive as it was not the point of the test anyway